crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -201,6 +201,22 @@ def compute_resources_available?
         true
       end
 
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
```

Comment:
The name is a little bit confusing. I would name it `ha_present?` and return a true or false value.